xline/src/server/xline_server.rs

```diff
@@ -390,17 +397,36 @@ impl XlineServer {
                 watcher,
                 Arc::clone(&header_gen),
                 *self.server_timeout.watch_progress_notify_interval(),
+                self.shutdown_trigger.subscribe(),
             ),
             MaintenanceServer::new(persistent, header_gen),
             curp_server,
         ))
     }
+
+    /// Check if `XlineServer` is stopped
+    #[inline]
+    #[must_use]
+    pub fn is_stopped(&self) -> bool {
+        self.shutdown_trigger.is_closed()
+    }
+
+    /// Stop `XlineServer`
+    #[inline]
+    pub async fn stop(&self) {
+        self.shutdown_trigger.self_shutdown_and_wait().await;
+    }
 }
 
 impl Drop for XlineServer {
     #[inline]
     fn drop(&mut self) {
-        self.shutdown_trigger.notify(usize::MAX);
-        self.curp_shutdown_trigger.self_shutdown();
+        if !self.is_stopped() {
+            let task_number = self.shutdown_trigger.receiver_count();
+            error!(
```

Comment:
Could we also utilize tokio's [`RuntimeMetrics` ](https://docs.rs/tokio/latest/tokio/runtime/struct.RuntimeMetrics.html) to ensure every task is properly exit.

Reply:
This is a unstable feature of tokio